dcos/rpcclient.py

```diff
@@ -145,7 +144,8 @@ def http_req(self, method_fn, path, *args, **kwargs):
                 request_method=e.response.request.method,
                 request_url=e.response.request.url,
                 json_body=json_body)
-            raise DCOSException(message)
+
+            raise DCOSException(message) from e
```

Comment:
Throwing a new exception hides the original error and its stack trace. `from e` adds a `__cause__` and the stack trace. See exception chaining in [PEP-3134](https://www.python.org/dev/peps/pep-3134/).